halo2_proofs/src/plonk/circuit.rs

```diff
@@ -1352,6 +1353,20 @@ impl<F: Field> Mul<F> for Expression<F> {
     }
 }
 
+impl<F: Field> Sum<Self> for Expression<F> {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.reduce(|acc, x| acc + x)
+            .unwrap_or(Expression::Constant(F::ZERO))
+    }
+}
+
+impl<F: Field> Product<Self> for Expression<F> {
+    fn product<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.reduce(|acc, x| acc * x)
+            .unwrap_or(Expression::Constant(F::ONE))
```

Comment:
Returning `1` makes more sense to me, since it's multiplicative identity. Also most field implementations return `1` on empty product, for instance https://github.com/zcash/pasta_curves/blob/a586b8c2db5ff9276cc1c2b4260f99ae68eeff02/src/fields/fq.rs#L176-L186.

Reply:
Fair point! Let's get it merged then! 

Thanks a lot for the reference!